### PR TITLE
re-enable jdk9 collections support.

### PR DESCRIPTION
**Pull request by SimY4**

### CODE REVIEW

File: value-fixture/test/org/immutables/fixture/jackson/ObjectMappedTest.java [hunk 1]
```diff
@@ -27,7 +28,8 @@
 public class ObjectMappedTest {
   private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
   {
-    OBJECT_MAPPER.registerModule(new GuavaModule());
+    OBJECT_MAPPER.registerModule(new GuavaModule())
+        .enable(SerializationFeature.ORDER_MAP_ENTRIES_BY_KEYS);
```
Comment (SimY4) [inline value-fixture/test/org/immutables/fixture/jackson/ObjectMappedTest.java:32]: To make serialization stable between JDK8 and JDK9+.

File: value-fixture/test/org/immutables/fixture/jdkonly/JdkOnlyTest.java [hunk 2]
```diff
@@ -48,10 +48,10 @@ public void collections() {
...
         .build();
 
-    check(coll.ints()).isOf(1, 2, 3, 4, 5, 6);
-    check(coll.navs()).isOf(3, 2, 1);
-    check(coll.ords()).isOf(4, 5, 6, 7, 8, 9);
-    check(coll.pols()).isOf(RetentionPolicy.RUNTIME);
+    check(coll.ints()).hasContentInAnyOrder(1, 2, 3, 4, 5, 6);
+    check(coll.navs()).hasContentInAnyOrder(3, 2, 1);
+    check(coll.ords()).hasContentInAnyOrder(4, 5, 6, 7, 8, 9);
+    check(coll.pols()).hasContentInAnyOrder(RetentionPolicy.RUNTIME);
```
Comment (SimY4) [inline value-fixture/test/org/immutables/fixture/jdkonly/JdkOnlyTest.java:54]: To make comparisons stable between JDK8 and JDK9+.
`isOf` was taking verifying elements is exact order

**Comment by SimY4** [comment]
Hang on, let me check something

**Comment by SimY4** [comment]
Good to go. 👍

**Comment by SimY4** [comment]
@elucash @asereda-gs may I bring your attention to this one? It re-enables JDK9 collections use.

**Comment by elucash** [comment]
@SimY4 thank you for the PR! sorry for the late response, usual cycle of work/business-trip/in-the-bed-with-flue etc. While I think that we can re-enable JDK9 support (way overdue, right), it's still somewhat of a breaking change, unfortunately: those subtleties with `Map`/`Set` ordering and null elements. Can we add a style flag (sign... not happy about it either) and use it in `isGenerateJdk9()` ?

**Pull request merged**